### PR TITLE
Run mdbook and update GitHub pages automatically upon every push

### LINKED ISSUE #2: Create GitHub action that runs mdbook and updates GitHub pages automatically upon every push

### PR DESCRIPTION
**Pull request by shepmaster**
Closes #2

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Build and Deploy
+on:
+  push:
+    branches:
+      - master
```
Comment (shepmaster) [inline .github/workflows/publish.yml:5]: We probably also wan this to *run* the build on every PR, but not publish.

**Comment by shepmaster** [comment]
If you are comfortable using the `JamesIves/github-pages-deploy-action` action, we should comment out the `if` line, push this branch again, make sure that it deploys correctly, then restore the `if`.

**Comment by dylanmckay** [comment]
Nice work @shepmaster!!

I've updated the `if` as you suggest in da4f27a951d43ffc55995bb5164f392046b32d2e to test it. We will see how the deploy goes.

**Comment by dylanmckay** [comment]
> rsync: change_dir "/home/runner/work/book.avr-rust.com/book.avr-rust.com//build" failed: No such file or directory (2)

**Comment by shepmaster** [comment]
Alright; that's fixed. We can probably rebase out the branch/`if` hacks and squash down the changes, then merge. I'll let you do that to avoid conflicting with your manual pushes.

**Pull request merged**